src/JSInterop/Microsoft.JSInterop/src/Infrastructure/ByteArrayJsonConverter.cs

```diff
@@ -0,0 +1,91 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Buffers;
+using System.Text.Json;
+using System.Text.Json.Serialization;
+using System.Threading;
+
+namespace Microsoft.JSInterop.Infrastructure
+{
+    internal sealed class ByteArrayJsonConverter : JsonConverter<byte[]>
+    {
+        // Atomically incrementing unique identifier for serializing byte arrays.
+        private static long _byteArrayId;
```

Comment:
Can we make this non-static, and stored per-`JSRuntime` instance?

The issue with it being static is that it discloses information about other users' activities. We try to keep all IDs scoped to be per-connection (or GUIDs) so that people can't infer things about the traffic level on the site, how long the server has been up, etc.

Reply:
Good catch